src/app/components/List/EBookList.jsx

```diff
@@ -17,27 +19,84 @@ import PropTypes from 'prop-types';
  * @return {string}
  */
 const generateLink = (url, eReaderUrl, local, download, ebook) => {
-  const link = (local && ebook && !download) ? `${eReaderUrl}?url=${url}` : `${url}`;
-  const linkText = (ebook && !download) ? 'Read Online' : 'Download';
+  const link = local && ebook && !download ? `${eReaderUrl}?url=${url}` : `${url}`;
+  const linkText = ebook && !download ? 'Read Online' : 'Download';
   return <a href={`${link}`}>{`${linkText}`}</a>;
 };
 
+// generates an array of options for the dropdown
+const generateOption = (link, eReaderUrl) => {
+  const url = link.local && link.ebook && !link.download ? `${eReaderUrl}?url=${link.url}` : `${link.url}`;
+  const label = link.label;
+  let className;
+  if (!link.local) {
+    className = 'external';
+  }
+  return {
+    value: label,
+    url,
+    label,
+    title: url,
+    ariaLabel: label,
+    className,
+  };
+};
+
+const onSelectChange = (value, options) => {
+  const match = options.find(option => option.label === value);
+  if (match) {
+    global.window.location.href = match.url;
+  }
+};
+
+// gets all links from all ebooks, classified on download true or false
+const linksArray = ({ ebooks, download }) => _flatten(ebooks.map(item => item.links)).filter(link => link.download === download);
+
+// render the dropdown and/or the link to the ebook, classifed by download true or false
+const LinksSelector = ({ ebooks, download, eReaderUrl }) => {
+  const linksList = linksArray({ ebooks, download });
+  const options = linksList.map(link => Object.assign({}, generateOption(link, eReaderUrl)));
+  return (
+    <ul className="nypl-ebooks-list">
+      {linksList.length > 1 && (
+        <li className="ebooks-list-dropdown">
+          <Dropdown
+            options={options}
+            setSelected={e => onSelectChange(e, options)}
+            placeholder={download ? 'Download' : 'Read Online'}
+            selectedOption=""
+            buttonClassName=""
+            hideArrow
+          />
+        </li>
+      )}
+      {linksList.length === 1
+        && linksList.map((link, linkKey) => (
+          <li key={`${linkKey.toString()}`}>{generateLink(link.url, eReaderUrl, link.local, link.download, link.ebook)}</li>
+        ))}
+    </ul>
+  );
+};
+
 const EBookList = ({ ebooks, eReaderUrl }) => (
-  <ul>
-    {ebooks.map((item, ebookKey) => (
-      <li key={`${ebookKey.toString()}`}>
-        <ul className="nypl-ebooks-list">
-          {item.links.map((link, linkKey) => (
-            <li key={`${ebookKey.toString()}-${linkKey.toString()}`}>
-              {generateLink(link.url, eReaderUrl, link.local, link.download, link.ebook)}
-            </li>
-          ))}
-        </ul>
-      </li>
-    ))}
+  <ul className="nypl-ebooks-list">
+    <li>{LinksSelector({ ebooks, download: true, eReaderUrl })}</li>
+    <li>{LinksSelector({ ebooks, download: false, eReaderUrl })}</li>
```

Comment:
Would it be easier to filter the the ebooks here for the `download` value?

Reply:
the download value is needed for the text in the buttons